### PR TITLE
BSbodybags inherit heaviest item weight + Easier to escape the disarm/BSbodybag combo

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
closes #53760
closes #53747

## About The Pull Request

Bodybags now inherit the weight of their heaviest item. If it doesn't have a weight (people), we assume bulky since things like people can be firecarried or whatever.

It now takes less time to get out of a bag (20 > 12 secs), and the progress bar is unbreakable unless you get incapacitated. This means people interested in using it to kidnap can still do so but it requires considerably more effort that isnt based on a latency difference and a quick disarm. The timer still runs down if you open the bag and try to put them back in but good attempt:smirk:.

This also fixes an issue with the BSB where it was picking up or otherwise doublecalling checks from inheritance.

## Why It's Good For The Game

I liked aspects/concepts of both of their respective PRs so I decided to pull from each. This PR attempted to fix the real concerns with the item without entirely killing some of the neat uses it has.

Fix is fix

## Changelog
:cl: Cobby (Ideas pooled from Anne and ATH)
balance: BSbodybags now inherit the weight of its heaviest item.
balance: It is much easier for a <i>conscious</i> individual to escape the bluespace bodybag.
/:cl:

### CODE REVIEW

**Comment by zxaber** [comment]
Good compromise, even if them being bulky means we can't use them to play assistant pokemon.

**Comment by ExcessiveUseOfCobblestone** [comment]
bulky can still be picked up (and technically so can gigantic i think), it just means you cant pocket/backpack it

**Comment by XDTM** [comment]
With these mechanics you should probably rename it as 'sack' and make the sprite grow larger with the contents' size. The bluespace part that makes it bigger inside is not really doing anything if a large item makes the container just as large, so we might as well embrace it and allow people to stuff others in sacks over their shoulders to kidnap them.

**Comment by necromanceranne** [comment]
Just call it a 'portable hole', to keep it on theme with ripping off the extradimensional storage objects from DnD.

(Also have it go boom if you put one interdimensional bag into another, just like the real portable hole and bags of holding do)

**Comment by XDTM** [comment]
Really though, i think that a lot of the friction in expectations for this item is because it was born as a bodybag upgrade and the folding functionality was tacked on afterwards.
In my opinion the current version could be restyled into a bluespace hypercube or something and left as-is (maybe with the tweaks to avoid push-kidnapping) and placed somewhere higher in the bluespace techweb tree, instead of these nerfs which preclude a _lot_ of the possible creative uses of storing mobs in a small item.

**Comment by ExcessiveUseOfCobblestone** [comment]
It can still hold unholy amounts of stuff and you can still pickup this item without difficulties. It's no longer a direct reference but its a bit silly to suggest the reference is lost if you are remotely affected by the contents you put in it.

The clown car effect is still a part of the bag, it's just that you cant pocket it or backpack it.

**Comment by ATH1909** [comment]
I still feel like making resisting out of the bag instant if its loc() isn't a person (but making normal resisting out of the bag still interruptable) would be the best idea here.

Still, I think you're making much more progress with this PR than I ever will with mine, so I think I'll just close my PR.

File: code/game/objects/items/bodybag.dm [hunk 1]
```diff
@@ -77,8 +77,13 @@
 	user.last_special = world.time + CLICK_CD_BREAKOUT
 	to_chat(user, "<span class='notice'>You claw at the fabric of [src], trying to tear it open...</span>")
 	to_chat(loc, "<span class='warning'>Someone starts trying to break free of [src]!</span>")
-	if(!do_after(user, 200, target = src))
-		to_chat(loc, "<span class='warning'>The pressure subsides. It seems that they've stopped resisting...</span>")
-		return
-	loc.visible_message("<span class='warning'>[user] suddenly appears in front of [loc]!</span>", "<span class='userdanger'>[user] breaks free of [src]!</span>")
-	qdel(src)
+	if(do_after_mob(user, src, 12 SECONDS, TRUE))
```
Comment (ATH1909) [inline code/game/objects/items/bodybag.dm:80]:
```suggestion
	if(!isliving(src.loc) || do_after(user, 10 SECONDS, target = src))
```
This way, you can still use it for kidnapping, but you can't cremate someone with no possible counterplay for them to escape (aside from, say, putting the bluespace body bag in the hands of a monkeyman or something, which you could maybe solve with a client check).

Comment (ATH1909) [reply code/game/objects/items/bodybag.dm:80]: Oh, and I guess maybe a check to keep people from just chucking a bluespace body bag that contains someone into a recycler would also be nice.

**Comment by ATH1909** [comment]
> Edit: As an addendum, since kidnapping without restraints becomes near-impossible after this change,

From what I can tell, under this change, putting restraints on someone wouldn't help (unless that makes them count as being incapacitated?), as a character in the bag who's resisting try to break out of the bag before trying to break out of their restraints, I think.

File: code/game/objects/items/bodybag.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/items/bodybag.dm:80]: it doesnt make sense to me that being trapped several layers in somehow makes it easier to escape than being trapped while in someones hands.

File: code/game/objects/structures/crates_lockers/closets/bodybag.dm [hunk 2]
```diff
@@ -37,11 +37,11 @@
 		if(!user.canUseTopic(src, BE_CLOSE))
 			return
 		if(t)
-			name = "body bag - [t]"
+			name = "[initial(name)] - [t]"
 			tagged = TRUE
 			update_icon()
 		else
-			name = "body bag"
+			name = "[initial(name)]"
```
Comment (ShizCalev) [inline code/game/objects/structures/crates_lockers/closets/bodybag.dm:44]:
```suggestion
			name = initial(name)
```

**Review by ShizCalev — CHANGES_REQUESTED** [review]

**Comment by Farquaar** [comment]
Am I the only person who actually uses body bags for transporting dead bodies?

**Comment by Kromgar** [comment]
I love this PR as I was one of the people that pushed for Bluespace Body Bags to be able to hold actual people a long time ago... iirc it was Xhuis who implemented it on my suggestion. I would hate to see the kidnapping ability of the bodybag to go away.

File: code/game/objects/structures/crates_lockers/closets/bodybag.dm [hunk 3]
```diff
@@ -91,25 +112,37 @@
...
+	var/obj/item/bodybag/B = foldedbag_instance || new foldedbag_path
+	var/max_weight_of_contents = initial(B.w_class)
+	for(var/atom/movable/A in contents)
+		A.forceMove(B)
+		if(isliving(A))
+			to_chat(A, "<span class='userdanger'>You're suddenly forced into a tiny, compressed space!</span>")
+		if(!isitem(A))
+			max_weight_of_contents = max(WEIGHT_CLASS_BULKY,max_weight_of_contents)
+			continue
+		var/obj/item/A_is_item = A
```
Comment (Rohesie) [inline code/game/objects/structures/crates_lockers/closets/bodybag.dm:143]:
```suggestion
	for(var/am in contents)
		var/atom/movable/content = am
		content.forceMove(B)
		if(isliving(content))
			to_chat(content, "<span class='userdanger'>You're suddenly forced into a tiny, compressed space!</span>")
		if(!isitem(content))
			max_weight_of_contents = max(WEIGHT_CLASS_BULKY, max_weight_of_contents)
			continue
		var/obj/item/A_is_item = content
```

**Pull request merged**